src/Algebra/Graph/Labelled/AdjacencyMap/Algorithm/Dijkstra.hs

```diff
@@ -0,0 +1,109 @@
+module Algebra.Graph.Labelled.AdjacencyMap.Algorithm.Dijkstra where
+
+import Data.Map.Strict (Map, (!))
+import Algebra.Graph.Label (Semiring(..), (<+>), zero)
+import qualified Data.Map.Strict as Map
+import Algebra.Graph.Labelled.AdjacencyMap.Internal (AdjacencyMap(..))
+
+-- | Psuedocode for Generic-Single-Source-Shortest-Distance (G,s)
```

Comment:
No need to provide pseudocode of a classic algorithm in Haddock, better link to the Wikipedia and the paper you referred to previously. Users need a high-level description and examples.

The text you have is still useful, so don't delete it, but move to a separate standalone comment: it is useful for *developers*, not users of the library.